### DISCUSSION TITLE
AES 256 cipher to plaintext

### DISCUSSION THREAD

**Discussion by csxdotcc** [micropython/micropython#16255] [Using MicroPython]
how to cipher a plaintext with a password in php 
and decipher in micropython?

php openssl-encrypt example: [php.net](https://www.php.net/manual/en/function.openssl-encrypt.php#example-942)

i've tried ucryptolib and also this lib [mpyaes](https://github.com/iyassou/mpyaes) encountered many many obsticles from hex en/decoding, mismatching blocksizes and ValueErrors. tried for multiple days now, cGPT is even more confused than i am xD

any ideas on how to bring this together? or anything ready to go i haven't found oO

**Comment by bidrohini** [comment]
You can get some clues for here: https://stackoverflow.com/questions/54176838/encrypt-text-in-php-and-decrypt-in-python

**Comment by rkompass** [comment]
It should be doable as [this](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://wokwi.com/projects/321484545174012499&ved=2ahUKEwilz6rozeCJAxVtR_EDHVKBK8sQFnoECBQQAQ&usg=AOvVaw09s9eT728fui4PfWlMWvLt) example shows.

If one side, e.g. encryption, is wrong, you may perhaps try forever to get the other side right.
